src/app/game-state/activity.service.ts

```diff
@@ -133,6 +133,10 @@ export class ActivityService {
         const rest = this.getActivityByType(ActivityType.Resting);
         if (!this.checkResourceUse(activity) && rest.unlocked && this.autoRestUnlocked){ // check for resources, rest activity is available, and autopause unlocked.
           activity = rest;
+          if (this.checkMaxResourceUse(activity)) {
+            //don't skip the current activity unless you don't have enough max resources to fit it
```

Comment:
Hmm. Not sure if I like this better. This would stall you at this activity until you've rested enough to do it, right? Thought @Hornsoup ?

Reply:
This would actually do nothing, since the activity would change to rest and then have it checked for the 0 requirements for rest. None of this code is useful.

Reply:
Actually, yeah, with the later currentTickCount reduction that would just rest without passing days IIUC?

Reply:
I don't think this messes with passing days, though that was my first instinct when I looked at it. Then I remembered that variable just counts how far along in the repeats for that activityLoop entry we are. Days still pass, we just don't advance to the next activity until we've rested enough to do this one (unless we'll never be able to, in which case it skips with the maxResource check).

Reply:
Ok, I think I understand what's tricking me. It's the second maxresourceuse check after the rest. It will always reduce the currenttickcount, causing it to potentially get stuck at 0.